app/src/main/java/org/docspell/docspellshare/http/UploadManager.java

```diff
@@ -46,8 +48,10 @@ public void submit(HttpRequest request) {
     public void run() {
       Process.setThreadPriority(Process.THREAD_PRIORITY_BACKGROUND);
       try {
-        int code = request.execute(listener);
-        listener.onFinish(code);
+        Response resp = request.execute(listener);
+        listener.onFinish(resp.code());
+        resp.close();
```

Comment:
Would it make sense to have another `try-finally` for the closing here?

Reply:
You could add this, yes.